contribs/gmf/src/directives/layertree.js

```diff
@@ -268,6 +269,120 @@ gmf.LayertreeController.prototype.getNodeState = function(treeCtrl) {
 
 
 /**
+ * Get the icon image URL for the given treeCtrl's layer. It can only return a
+ * string for internal WMS layers without multiple childlayers in the node.
+ * @param {ngeo.LayertreeController} treeCtrl ngeo layertree controller, from
+ *     the current node.
+ * @return {?string} The icon legend URL or null.
+ * @export
+ */
+gmf.LayertreeController.prototype.getLegendIconUrl = function(treeCtrl) {
+  var layer = treeCtrl.layer;
+  var node = treeCtrl.node;
+
+  if (goog.isDef(node.children) ||
+      node.type === 'WMTS' ||
+      node.type === 'external WMS' ||
+      goog.isDef(node.childLayers) && node.childLayers.length > 1) {
+    return null;
+  }
+
+  goog.asserts.assertInstanceof(layer, ol.layer.Image);
+  var opt_legendRule = node.metadata['legend_rule'];
+  return this.getWMSLegendUrl_(layer, opt_legendRule);
+};
+
+
+/**
+ * Get the complete legend URL for the given treeCtrl's layer.
+ * @param {ngeo.LayertreeController} treeCtrl ngeo layertree controller, from
+ *     the current node.
+ * @return {?string} The legend URL or null.
+ * @export
+ */
+gmf.LayertreeController.prototype.getLegendUrl = function(treeCtrl) {
+  var layer = treeCtrl.layer;
+  var node = treeCtrl.node;
+
+  if (goog.isDef(node.children)) {
+    return null;
+  }
+
+  if (node.type === 'WMTS') {
+    goog.asserts.assertInstanceof(layer, ol.layer.Tile);
+    return this.getWMTSLegendUrl_(layer);
+  } else {
+    goog.asserts.assertInstanceof(layer, ol.layer.Image);
+    return this.getWMSLegendUrl_(layer);
+  }
+};
+
+
+/**
+ * Get the WMTS legend URL for the given layer.
+ * @param {ol.layer.Tile} layer Tile layer as returned by the
+ * gmf layerHelper service.
+ * @return {?string} The legend URL or null.
+ * @private
+ */
+gmf.LayertreeController.prototype.getWMTSLegendUrl_ = function(layer) {
+  // FIXME case of multiple styles ?  case of multiple legendUrl ?
```

Comment:
I don't have any example of WMTS layers with multiple styles and, or, legend URL. So I don't know what I must do in this case...